test_requests.py

```diff
@@ -411,6 +416,21 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
+    def test_DIGESTAUTH_THREADED(self):
```

Comment:
No, currently there are no tests to confirm that requests is thread-safe. Such testing is really tricky without explicit language support. However, this lack of testing is why I believe that currently requests is not *truly* thread-safe.

Again, think carefully about whether you can use `threading.Event` to coordinate two threads in such a way that you can deterministically break them (e.g. by having them get their nonces confused).